lib/encoding.neon

```diff
@@ -0,0 +1,159 @@
+%|
+ |  File: encoding
+ |
+ |  Functions for encoding and decoding ASCII or Binary data in various formats.
+ |%
+IMPORT bitwise
+
+EXPORT base64Encode
+EXPORT base64Decode
+
+TYPE Mode IS ENUM
```

Comment:
I'm not sure having an enum for this is terribly helpful. First, terminology - it's not a "mode", but a "state" (as in "state machine"). Second, this would be equivalently served by a simple number. Third, you don't actually even have to keep track since in your loop you already know what offset you're working with, so you can tell what you're doing with just a MOD 3.